### PR TITLE
feat: fail properly if predeploy fails

### PR DESCRIPTION
**Pull request by csaunders-ldt**
This jq is your fault @LukeSheard. It's entirely your fault.

(use an `if` statement to conditionally call predeploy rather than just eating errors. npm supports `--if-present` but yarn doesn't, so we have to do this crap)

### CODE REVIEW

File: .github/workflows/deploy-production.yml [hunk 1]
```diff
@@ -79,7 +79,7 @@ jobs:
         with:
           github-token: ${{ secrets.WORKFLOW_CALL_TOKEN }}
       - uses: stampedeapp/actions/setup@main
-      - run: LDT_ENV=staging yarn predeploy || true
+      - run: LDT_ENV=staging if [ "$(cat package.json | jq -r .scripts.predeploy)" != "null" ]; then yarn predeploy; fi;
         env:
           MONGO_CONNECTION_STRING: ${{ secrets.MONGO_CONNECTION_STRING }}
```
Comment (LukeSheard) [inline .github/workflows/deploy-production.yml:84]:
```suggestion
```suggestion
      - id: predeploy
        name: Check for predeploy script
        run: echo "script=$(cat package.json | jq -r .scripts.predeploy)" >> $GITHUB_OUTPUT
      - run: ${{ steps.predeploy.outputs.script }}
        if: ${{ steps.predeploy.outputs.script }}
       name: Run predeploy script
        env:
          LDT_ENV: staging
          MONGO_CONNECTION_STRING: ${{ secrets.MONGO_CONNECTION_STRING }}
```

**Review by trevorah — APPROVED** [review]
I'm happy with either option, but I agree with luke that this can be more readable.

File: predeploy/action.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      if: ${{ inputs.github-token !== '' }}
+      with:
+        github-token: ${{ inputs.github-token }}
+    - uses: stampedeapp/actions/setup@main
+    - id: predeploy
+      name: Check for predeploy script
+      shell: bash
+      run: echo "script=$(cat package.json | jq -r .scripts.predeploy)" >> $GITHUB_OUTPUT
+    - shell: bash
+      if: ${{ steps.predeploy.outputs.script }}
```
Comment (LukeSheard) [inline predeploy/action.yml:28]: I think you need to compare this to `"null"` ?

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -58,8 +58,9 @@ jobs:
     concurrency:
       group: cloudformation-dev
     steps:
-      - uses: stampedeapp/actions/setup@main
-      - run: LDT_ENV=dev yarn predeploy || true
+      - uses: stampedeapp/actions/predeploy@wip
```
Comment (LukeSheard) [inline .github/workflows/deploy.yml]:
```suggestion
      - uses: stampedeapp/actions/predeploy@main
```

File: predeploy/action.yml (hunk 2 above)
Comment (csaunders-ldt) [reply predeploy/action.yml]: yeah, I copied your code above and didn't verify that you got it right 😂

**Pull request merged**